internal/versions/api.go

```diff
@@ -77,62 +76,67 @@ func GetModelVersions(v Versions) models.Versions {
-		currentMajorMinorVersion, err := common.GetMajorMinorVersion(*releaseImage.OpenshiftVersion)
+		isMajorMinorEqual, err := common.BaseVersionEqual(*releaseImage.Version, majorMinorVersion)
 		if err != nil {
 			return nil, err
 		}
 
-		if *currentMajorMinorVersion != majorMinorVersion {
+		if !isMajorMinorEqual {
 			continue
 		}
 
-		if latestReleaseImage == nil {
-			latestReleaseImage = releaseImage
+		if latestReleaseImages == nil {
+			latestReleaseImages = models.ReleaseImages{releaseImage}
 		}
 
-		isNewest, err := common.BaseVersionLessThan(*releaseImage.Version, *latestReleaseImage.Version)
+		isEqual, err := common.MajorMinorPatchEqual(*releaseImage.Version, *latestReleaseImages[0].Version)
+		if err != nil {
+			return nil, err
+		}
+		if *isEqual && !lo.Contains(latestReleaseImages, releaseImage) {
+			latestReleaseImages = append(latestReleaseImages, releaseImage)
+			continue
+		}
+
+		isNewest, err := common.BaseVersionLessThan(*releaseImage.Version, *latestReleaseImages[0].Version)
 		if err != nil {
 			return nil, err
 		}
 
 		// none-beta > beta, later-beta > beta
-		if latestReleaseImage.SupportLevel == models.OpenshiftVersionSupportLevelBeta {
+		if latestReleaseImages[0].SupportLevel == models.OpenshiftVersionSupportLevelBeta {
 			if isNewest || releaseImage.SupportLevel != models.OpenshiftVersionSupportLevelBeta {
-				latestReleaseImage = releaseImage
+				latestReleaseImages = []*models.ReleaseImage{releaseImage}
 			}
 		} else { // non-beta-later > non-beta
 			if isNewest && releaseImage.SupportLevel != models.OpenshiftVersionSupportLevelBeta {
-				latestReleaseImage = releaseImage
+				latestReleaseImages = []*models.ReleaseImage{releaseImage}
 			}
 		}
-	}
-
-	log.Debugf("Found latest release image version '%s' for majorMinorVersion '%s'", *latestReleaseImage.Version, majorMinorVersion)
 
-	return latestReleaseImage, nil
-}
+	}
 
-// getMultiArchReleaseImage retrieves multi arch release image matching a given x.y.z version.
-func getMultiArchReleaseImage(releaseImages models.ReleaseImages, version string, log logrus.FieldLogger) *models.ReleaseImage {
-	if !strings.HasSuffix(version, "-multi") {
-		version = fmt.Sprintf("%s-multi", version)
+	if len(latestReleaseImages) == 0 {
+		return nil, errors.Errorf("No release image found for version '%s'", majorMinorVersion)
 	}
 
-	if releaseImage, ok := lo.Find(releaseImages, func(releaseImage *models.ReleaseImage) bool {
-		return swag.StringValue(releaseImage.Version) == version
-	}); ok {
-		log.Debugf("Found multi-arch release image for version '%s'", version)
-		return releaseImage
+	log.Debugf(
+		"Found latest release images version '%s' for majorMinorVersion '%s' with CPU architectures: ",
+		*latestReleaseImages[0].Version,
+		majorMinorVersion,
+	)
+	for _, releaseImage := range latestReleaseImages {
+		log.Debug(*releaseImage.CPUArchitecture)
```

Comment:
Build a unique string instead, and log all at once, so the there will be all be information on the same line, and there will be no risk to get lines mixed with other logs.
Also, it may not be the case here, but some log lines may be discarded by the logger if it implements a rate-limiter to avoid log flooding.